### ISSUE TITLE
Display first seen/sent time in transaction details

### ISSUE THREAD

**Issue by leafcutterant** [spesmilo/electrum#6165]
I propose displaying in the transaction details the date/time of when it was sent or first seen. The rationale is to provide better ways to track the progression of transactions, even when they are not yet confirmed, as the 10 minute block time and congestions create a considerable gap between transaction intent and confirmation. 

Proposed logic:

* unconfirmed transaction:
  * if sent from the client: date/time of sending
  * if received from the network: date/time of first seeing it
* confirmed transaction: display the time/date of both the inclusion and sending/first seeing it

**Comment by gits7r** [comment]
How do you propose to re-build this information upon wallet restoration? Say you restore your wallet from seed, this data (first seen) is nowehere to be fetched from. So it will not work from past transactions when wallet is restored. I don't see a down side for having this as well, maybe it's useful from some use cases, however for transactions in history that are restored this field will be a `N/A`. 

Or am I missing something and is there a way to get this info from somewhere upon restoration from seed?

Also, say you have the same wallet on more computers. Work PC, laptop and mobile phone.
You send a transaction from your work laptop (that will have the first seen date as the timestamp when you send it). You turn on your travel laptop 3 days later, and that transaction is long time already confirmed. This cannot have as well the fist seen timestamp.

The time and code to implement this might not be worth it, particularly for this cases. Electrum is designed to be a light weight easy to rebuild from seed wallet -- the more features we add that cannot be reproduced deterministically from seed or from blockchain data, the worse it is.

**Comment by leafcutterant** [comment]
> Or am I missing something and is there a way to get this info from somewhere upon restoration from seed?

No, this would be locally stored data that can't be restored from seed. Electrum has plenty of such data, e.g. transaction descriptions, address labels or custom fiat values. These are not restorable from the blockchain yet they are useful. (Historical fiat values aren't on the blockchain either, though one could argue there's relative consensus about the price among the different sources.)

> however for transactions in history that are restored this field will be a N/A.

> You turn on your travel laptop 3 days later, and that transaction is long time already confirmed. This cannot have as well the fist seen timestamp.

Good observations! I'll update the proposed logic to account for these.
